src/services/HelpOfferService.js

```diff
@@ -49,53 +50,105 @@ class OfferedHelpService {
-
-    if(isUserEntity){
-      userPosition = helpOffer.possibleEntities.indexOf(helpedId);
-      user = this.EntityService;
-      findOneUser = "findOneEntityWithProjection";
-      possibleHelpedUser = helpOffer.possibleEntities;
-    } else {
-      userPosition = helpOffer.possibleHelpedUsers.indexOf(helpedId);
-      user = this.UserService;
-      findOneUser = "findOneUserWithProjection"; 
-      possibleHelpedUser = helpOffer.possibleHelpedUsers;
-    }
+  validateOwnerAndHelpedUser(helpedId, helpOffer) {
+    if (helpOffer.ownerId == helpedId)
+      throw new Error("Dono não pode ser ajudado da própria oferta");
+    else if (helpOffer.helpedUserId != null && helpOffer.helpedUserId.includes(helpedId))
+      throw new Error("Usuário já está sendo ajudado");
+  }
 
-    if(helpOffer.ownerId == helpedId){
-      throw new Error("Usuário não pode ser ajudante da própria oferta");
-    }
-    else if (userPosition > -1) {
+  isUserInPossibleHelpedUsers(helpedUser, helpOffer, helpedId) {
+    if (!helpedUser.isEntity) 
+     return helpOffer.possibleHelpedUsers.includes(helpedId)
+    else 
+      return helpOffer.possibleEntities.includes(helpedId)
+  }
+
+  possibleInterestedArray(helpOffer, helpedUser){
+    if(helpedUser.isEntity)
+      return helpOffer.possibleEntities
+    else
+      return helpOffer.possibleHelpedUsers
+  }
+
+  async addPossibleHelpedUsers(helpedId, helpOfferId) {
+    const helpOffer = await this.getHelpOfferById(helpOfferId);
+    const helpedUser = await this.verifyUserEntity(helpedId);
+    const possibleHelpedUser = this.possibleInterestedArray(helpOffer, helpedUser)
+
+    // Validacao
+    this.validateOwnerAndHelpedUser(helpedId, helpOffer)
+    if (this.isUserInPossibleHelpedUsers(helpedUser, helpOffer, helpedId))
       throw new Error("Usuário já é um possível ajudado");
-    }
 
-    await this.useService(possibleHelpedUser,"push",[helpedId]);
+    // Alteracao do array
+    await this.useService(possibleHelpedUser, "push", [helpedId]);
     await this.OfferedHelpRepository.update(helpOffer);
-    
-    const helpedUserProjection = { name: 1, _id: 0};
-    const { name: helpedUserName } = await this.useService(user,findOneUser,[helpedId,helpedUserProjection]);
-    
+
+
+    //Notificação
     const ownerProjection = { deviceId: 1, _id: 0 };
-    const { deviceId:ownerDeviceId } = await this.useService(this.UserService,"findOneUserWithProjection",[helpOffer.ownerId,ownerProjection]);  
-     
-    const title = `${helpedUserName} quer sua ajuda!`;
+    const { deviceId: ownerDeviceId } = await this.useService(this.UserService, "findOneUserWithProjection", [helpOffer.ownerId, ownerProjection]);
+
+    const title = `${helpedUser.name} quer sua ajuda!`;
     const body = `Sua oferta ${helpOffer.title} recebeu um interessado`;
 
+    await this.sendHelpOfferNotification(ownerDeviceId, title, body, ownerDeviceId, helpOfferId, notificationTypesEnum.ofertaRequerida);
```

Comment:
Aqui tá dando um erro:

![image](https://user-images.githubusercontent.com/54081877/133860972-319947c9-2830-40fa-89e2-a177f662f8fc.png)

Isso acontece porque está sendo passado o ownerDeviceId no lugar do ownerId
como tá:
`await this.sendHelpOfferNotification(ownerDeviceId, title, body,`**ownerDeviceId**`, helpOfferId, notificationTypesEnum.ofertaRequerida);`
Como deveria ser:
`await this.sendHelpOfferNotification(ownerDeviceId, title, body,`**ownerID**`, helpOfferId, notificationTypesEnum.ofertaRequerida);`

Aqui tem uma possivel solução para economizar o tempo de vossa senhoria: 
local: `services/HelpOfferServices.js>OfferedHelpService>addPossibleHelpedUsers`

<pre>
    //Notificação
    const ownerProjection = { deviceId: 1, _id: 1 };
    const owner = await this.useService(this.UserService, "findOneUserWithProjection", [helpOffer.ownerId, ownerProjection]);

    const title = `${helpedUser.name} quer sua ajuda!`;
    const body = `Sua oferta ${helpOffer.title} recebeu um interessado`;
    await this.sendHelpOfferNotification(owner.deviceId, title, body, owner._id, helpOfferId, notificationTypesEnum.ofertaRequerida);

</pre>

Reply:
talvez dê pra passar o proprio helpOffer.ownerId, como tava antes de refatorar

Reply:
Quê isso @thiagohdaqw , GOD dms, obg pela correção amigo ! E que mensagem formal ein kkkkkkkkkkkk amei.